fl4health/preprocessing/warmed_up_module.py

```diff
@@ -0,0 +1,121 @@
+            AssertionError(
+                "pretrained_model_path and pretrained_model is mutually exclusive. Please provide one of them."
+            )
+
+        elif pretrained_model is not None:
+            log(INFO, "Pretrained model is provided.")
+            self.pretrained_model_state = pretrained_model.state_dict()
+
+        elif pretrained_model_path is not None:
+            assert os.path.exists(
+                pretrained_model_path
+            ), f"Pretrained model path {pretrained_model_path} does not exist."
+            log(INFO, f"Loading pretrained model from {pretrained_model_path}")
+            self.pretrained_model_state = torch.load(pretrained_model_path).state_dict()
+
+        else:
+            raise AssertionError("At least one of pretrained_model_path and pretrained_model should be provided.")
+
+        if weights_mapping_path is not None:
+            with open(weights_mapping_path, "r") as file:
+                self.weights_mapping_dict = json.load(file)
+                log(INFO, f"Weights mapping dict: {self.weights_mapping_dict}")
+        else:
+            log(INFO, "Weights mapping dict is not provided. Matching stats directlly, based on current model's keys.")
+            self.weights_mapping_dict = None
+
+    def get_matching_component(self, key: str) -> Optional[str]:
+        """Get the matching component of the key from the weights mapping dictionary. Since the provided mapping
+        can contain partial names of the keys, this function is used to split the key of the current model and
+        match it with the partial key in the mapping, returning the complete name of the key in the pretrained model.
+
+        This allows users to provide one mapping for multiple statistics that share the same prefix. For example,
+        if the mapping is {"model": "global_model"} and the input key of the current model is "model.layer1.weight",
+        then the returned matching component is "global_model.layer1.weight".
+
+        Args:
+            key (str): Key to be matched in pretrained model.
+
+        Returns:
+            Optional[str]: If no weights mapping dict is provided, returns the key. Otherwise, if the key is in the
+            weights mapping dict, returns the matching component of the key. Otherwise, returns None.
+        """
+
+        if self.weights_mapping_dict is None:
+            return key
+
+        components = key.split(".")
+
+        for i, component in enumerate(components):
+            if i == 0:
+                matching_component = components[0]
+            else:
+                matching_component += "." + component
+            if matching_component in self.weights_mapping_dict:
+                return self.weights_mapping_dict[matching_component] + key[len(matching_component) :]
+        return None
+
+    def load_from_pretrained(self, model: torch.nn.Module) -> torch.nn.Module:
+        """Load the pretrained model into the current model.
```

Comment:
Super minor, but I might say `into a target model` since we provide a target into which, we insert the weights.